File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/DaysFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class DaysFunction : EndStartDateFunction
    {
        public const string Name = "DAYS";

        protected override string FunctionName => Name;

        private DaysFunction(Expression endDateExpression, Expression startDateExpression, SourceReference reference)
            : base(endDateExpression, startDateExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression endDateExpression,
            Expression startDateExpression) =>
            new DaysFunction(endDateExpression, startDateExpression, reference);
    }
}

File: typescipt/src/language/expressions/functions/daysFunction.ts


export class DaysFunction extends EndStartDateFunction {
   public const string Name = `DAYS`;

   protected override string FunctionName => Name;

   constructor(endDateExpression: Expression, startDateExpression: Expression, reference: SourceReference)
     : base(endDateExpression, startDateExpression, reference) {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression endDateExpression,
     Expression startDateExpression) {
     return new DaysFunction(endDateExpression, startDateExpression, reference);
   }
}
